File: spring-core-lab-solutions/Lab-05-solution/petclinic/src/test/java/com/petclinic/core/VisitServiceTest.java
package com.petclinic.core;

import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.context.ApplicationContext;

import static org.assertj.core.api.Assertions.*;

@SpringBootTest
public class VisitServiceTest {
    private final VisitService visitService;

    private final ApplicationContext applicationContext;

    @Autowired
    public VisitServiceTest(VisitService visitService, ApplicationContext applicationContext) {
        this.visitService = visitService;
        this.applicationContext = applicationContext;
    }

    @Test
    public void shouldFindVisit() {
        assertThat(this.visitService.findByReferenceNumber("V01-23").getReferenceNumber()).isEqualTo("V01-23");
    }

    @Test
    public void shouldDisplayBeanNumber() {
        System.out.println(this.applicationContext.getBeanDefinitionCount());
    }
}

File: spring-core-lab-solutions/Lab-kotlin-solution/src/test/kotlin/com/petclinic/core/VisitServiceTest.kt
package com.petclinic.core

import org.assertj.core.api.Assertions.*
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import java.math.BigDecimal
import java.time.LocalDate

@SpringBootTest
class VisitServiceTest(@Autowired val visitService: VisitService) {

    @Test
    fun shouldFindById() {
        val visit = visitService.findById(1)
        assertThat(visit.id).isEqualTo(1)
        assertThat(visit.pet.id).isEqualTo(1)
        assertThat(visit.owner.id).isEqualTo(1)
    }

    @Test
    fun shouldSaveVisit() {
        val pet = Pet(null, "hamster", "frosty")
        val owner = Owner(null, "Robert","Plant", BigDecimal(100), listOf(pet))
        val visit = Visit(null, LocalDate.now(), "teeth whitening", owner, pet)
        visitService.save(visit)
        assertThat(visit.id).isNotNull()
        assertThat(pet.id).isNotNull()
        assertThat(owner.id).isNotNull()
    }
}